dango/types/src/vesting.rs

```diff
@@ -0,0 +1,139 @@
+use {
+    anyhow::ensure,
+    grug::{
+        Addr, Coin, Duration, MultiplyFraction, NumberConst, Timestamp, Udec128, Uint128, Undefined,
+    },
+    std::collections::BTreeMap,
+};
+
+pub type ClaimablePosition = Position<Uint128>;
+
+#[grug::derive(Serde)]
+pub struct InstantiateMsg {}
+
+#[grug::derive(Serde)]
+pub enum ExecuteMsg {
+    CreatePosition {
+        user: Addr,
+        schedule: Schedule<Option<Timestamp>>,
+    },
+    Claim {
+        idx: u64,
+    },
+}
+
+#[grug::derive(Serde, QueryRequest)]
+pub enum QueryMsg {
+    #[returns(ClaimablePosition)]
+    Position { idx: u64 },
+    #[returns(BTreeMap<u64, ClaimablePosition>)]
+    Positions {
+        start_after: Option<u64>,
+        limit: Option<u32>,
+    },
+    #[returns(BTreeMap<u64, ClaimablePosition>)]
+    PositionsByUser {
+        user: Addr,
+        start_after: Option<u64>,
+        limit: Option<u32>,
+    },
+}
+
+#[grug::derive(Serde, Borsh)]
+pub struct Schedule<T = Timestamp> {
+    pub start_time: T,
+    pub cliff: Option<Duration>,
+    pub vesting: Option<Duration>,
+}
+
+impl Schedule<Option<Timestamp>> {
+    pub fn set_start_time(self, now: Timestamp) -> anyhow::Result<Schedule> {
+        let start_time = if let Some(start_time) = self.start_time {
+            ensure!(
+                start_time >= now,
+                "invalid start time: {:?}, now: {:?}",
+                start_time,
+                now
+            );
```

Comment:
This incorrect... Vesting start time isn't necessarily bigger than `now`. The starting time is the time that a contributor started working on the project (according to our contractor agreement) which is certainly earlier than `now` (which would be the chain genesis time).

Reply:
Yeah ok, this was in order to avoid typos on star_time. But if we need a start_time in the past the ensure can be removed